apps/bilateral_grid/filter.cpp

```diff
@@ -29,12 +29,29 @@ int main(int argc, char **argv) {
 
     // Timing code. Timing doesn't include copying the input data to
     // the gpu or copying the output back.
-    double min_t = benchmark(timing_iterations, 10, [&]() {
+
+    // Manually-tuned version
+    double min_t_manual = benchmark(timing_iterations, 10, [&]() {
         bilateral_grid(input, r_sigma, output);
     });
-    printf("Time: %gms\n", min_t * 1e3);
+    printf("Manually-tuned time: %gms\n", min_t_manual * 1e3);
+
+    // Auto-scheduled version
+    double min_t_auto = benchmark(timing_iterations, 10, [&]() {
+        bilateral_grid_auto_schedule(input, r_sigma, output);
+    });
+    printf("Auto-scheduled time: %gms\n", min_t_auto * 1e3);
 
     convert_and_save_image(output, argv[2]);
 
+    const halide_filter_metadata_t *md = bilateral_grid_metadata();
+    // Only compare the performance if target has non-gpu features.
+    if (!strstr(md->target, "cuda") &&
```

Comment:
The apps don't need to act as pass/fail performance tests. Can probably cut this section of code.